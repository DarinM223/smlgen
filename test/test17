datatype t = Cons of char * t list | Atom of char