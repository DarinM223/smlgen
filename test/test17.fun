functor Foo (structure S : FOO) = struct
  datatype t = Cons of char * t list | Atom of char
end