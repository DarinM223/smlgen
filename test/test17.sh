../smlgen --test test17.sml t:s test17.sig FOO.t:s test17.fun Foo.t:s test17 t:s
if cmp --silent test17.sml.test test17.sml.expected && cmp --silent test17.sig.test test17.sig.expected && cmp --silent test17.fun.test test17.fun.expected && cmp --silent test17.test test17.expected; then
  echo "test17 succeeded"
else
  echo "test17 failed"
  echo "Diff:"
  diff test17.sml.test test17.sml.expected
  diff test17.sig.test test17.sig.expected
  diff test17.fun.test test17.fun.expected
  diff test17.test test17.expected
  exit 1
fi