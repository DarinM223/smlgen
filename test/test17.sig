structure FOO = struct
  datatype t = Cons of char * t list | Atom of char
end