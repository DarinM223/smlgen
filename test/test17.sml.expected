datatype t = Cons of char * t list | Atom of char
local
  val rec t = fn t_0 =>
    fn Cons (t0, t1) =>
      "Cons "
      ^
      "("
      ^
      String.concatWith ", "
        [ "#\"" ^ Char.toString t0 ^ "\""
        , "[" ^ String.concatWith ", " (List.map t_0 t1) ^ "]"
        ] ^ ")"
     | Atom t2 => "Atom " ^ "(" ^ "#\"" ^ Char.toString t2 ^ "\"" ^ ")"
  val t = fn () => let val rec t_0 = fn ? => t t_0 ? in t_0 end
in val show = t ()
end